frontend/src/metabase/public/containers/PublicOrEmbeddedQuestion/PublicOrEmbeddedQuestion/PublicOrEmbeddedQuestion.tsx

```diff
@@ -157,7 +159,7 @@ export const PublicOrEmbeddedQuestion = ({
       console.error("error", error);
       dispatch(setErrorPage(error));
     }
-  }, [card, dispatch, parameterValues, token, uuid]);
+  }, [card, metadataRef, dispatch, parameterValues, token, uuid]);
```

Comment:
We cannot pass `metadata` here directly, otherwise we `run` will be executed on any metadata change

Reply:
Please add this as a comment in the code.